lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3707,16 +3707,45 @@ static void diagnoseOptionalToAnyCoercion(TypeChecker &TC, const Expr *E,
               .highlight(subExpr->getSourceRange())
               .fixItInsertAfter(subExpr->getEndLoc(), " as Any");
         }
-      }
+      } else if (auto *literal = dyn_cast<InterpolatedStringLiteralExpr>(E)) {
+        // Warn about interpolated segments that contain optionals.
+        for (auto &segment : literal->getSegments()) {
+          // Allow explicit casts.
+          if (auto paren = dyn_cast<ParenExpr>(segment)) {
+            if (isa<ExplicitCastExpr>(paren->getSubExpr())) {
+              continue;
+            }
+          }
 
+          if (auto segmentTy = segment->getType()) {
+            if (auto baseTy = segmentTy->getOptionalObjectType()) {
```

Comment:
In the spirit of http://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code, can you assign baseTy before the ``if`` and then invert the test and make it continue, unindenting the following code?